### DISCUSSION TITLE
Every so often xcodebuild runs when I use brew.

### DISCUSSION THREAD

**Discussion by ZenIsBestWolf** [Homebrew/discussions#3046] [Everyday usage] [answered]
This is causing significant slowdowns, and is becoming really annoying. I read that it was an issue with the bundled version of git, but I have the newer version of git installed from brew and it's what's being used (running `which git` outputs `/opt/homebrew/bin/git`) yet this still pops up frequently. How do I go about resolving this? I am on macOS Monterey 12.2.1 on an M1 Pro.

More information:

```
HOMEBREW_VERSION: 3.4.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ffc76654da12fba7d34d078fd9bf681fef65921e
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: af4413a13755c4f327f6894f2e8106dfd84ec796
Core tap last commit: 60 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_FORCE_BREWED_GIT: set
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.35.1 => /opt/homebrew/opt/git/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.2.1-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: 13.2.1
Rosetta 2: false
```

**Comment by carlocab** [comment]
@Bo98 would probably know about this.

**Comment by Bo98** [comment]
There's quite a few things which might invoke it, it's hard to to tell. Though it's usually not slow so something's wrong. When precisely does it occur? Does it happen when you do a `brew update`? What's the output of `brew config` (and is that slow too?)?

> I have the newer version of git installed from brew and it's what's being used

Note that you may have to set `HOMEBREW_FORCE_BREWED_GIT` for it to actually be used.

**Reply by ZenIsBestWolf** [reply]
This did not solve it, and it continues to run xcodebuild.

**Reply by carlocab** [reply]
It'll be really difficult for Bo to help you if you don't answer their questions.

**Reply by ZenIsBestWolf** [reply]
Sorry, I misread the message.

Here's my output of `brew config` (edit: I checked and YES this does cause `xcodebuild` to run)

```
HOMEBREW_VERSION: 3.4.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ffc76654da12fba7d34d078fd9bf681fef65921e
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: af4413a13755c4f327f6894f2e8106dfd84ec796
Core tap last commit: 60 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_FORCE_BREWED_GIT: set
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.35.1 => /opt/homebrew/opt/git/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.2.1-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: 13.2.1
Rosetta 2: false
```

**Comment by ZenIsBestWolf** [comment, marked as answer]
Alright well eventually I did find the solution, big thanks to [this stackoverflow post](https://stackoverflow.com/a/71639397)

I ran this then rebooted.
```sudo xcode-select -s /Library/Developer/CommandLineTools```
